test/suites/integration/daemonset_test.go

```diff
@@ -0,0 +1,111 @@
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	awstest "github.com/aws/karpenter/pkg/test"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var provider *v1alpha1.AWSNodeTemplate
+var provisioner *v1alpha5.Provisioner
+var limitrange *v1.LimitRange
+var priorityclass *schedulingv1.PriorityClass
+var daemonset *appsv1.DaemonSet
+var dep *appsv1.Deployment
+
+var _ = Describe("DaemonSet", func() {
+	BeforeEach(func() {
+		provider = awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+		}})
+		provisioner = test.Provisioner(test.ProvisionerOptions{
+			ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+		})
+		preemptNever := v1.PreemptNever
+		priorityclass = &schedulingv1.PriorityClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "high-priority-daemonsets",
+			},
+			PreemptionPolicy: &preemptNever,
+			Value:            int32(10000000),
+			GlobalDefault:    false,
+			Description:      "This priority class should be used for daemonsets.",
+		}
+		limitrange = &v1.LimitRange{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "limitrange",
+				Namespace: "default",
+			},
+			Spec: v1.LimitRangeSpec{
+				Limits: []v1.LimitRangeItem{{
+					Type: v1.LimitTypeContainer,
+					Default: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("2"),
+						v1.ResourceMemory: resource.MustParse("1Gi"),
+					}}},
+			},
+		}
+		daemonset = test.DaemonSet(test.DaemonSetOptions{
+			PodOptions: test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{Limits: v1.ResourceList{v1.ResourceMemory: resource.MustParse("1Gi")}},
+				PriorityClassName:    "high-priority-daemonsets",
+			},
+		})
+		numPods := 1
+		dep = test.Deployment(test.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "large-app"},
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceMemory: resource.MustParse("4")},
+				},
+			},
+		})
+	})
+	It("should account for LimitRange Containers Default For Resources", func() {
+		podSelector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+		daemonsetSector := labels.SelectorFromSet(daemonset.Spec.Selector.MatchLabels)
+		env.ExpectCreatedNodeCount("==", 0)
+		env.ExpectCreated(provisioner, provider, limitrange, priorityclass, daemonset, dep)
+		env.EventuallyExpectHealthyPodCount(podSelector, 1)
+		env.EventuallyExpectHealthyPodCount(daemonsetSector, 2)
+		env.ExpectCreatedNodeCount("==", 2)
```

Comment:
I'm wondering if we should take this a step further and assert that we are able to scheduled these pods on the same node. Maybe we consider enabling consolidation and ensure that we eventually converge to a single node?